### PR TITLE
Export a real name for vc to consume

### PR DESCRIPTION
**Pull request by Conan-Kudo**
Required for https://github.com/openSUSE/obs-build/pull/214

### CODE REVIEW

File: osc/commandline.py [hunk 1]
```diff
@@ -8675,7 +8675,20 @@ def do_vc(self, subcmd, opts, *args):
         else:
             apiurl = self.get_api_url()
 
-        # set user's email if the mailaddr env variable is not set
+        # set user's full name and email if the realname and mailaddr env variables are not set
+
+        if 'realname' in os.environ:
+            pass
+        elif 'realname' in conf.config['api_host_options'][apiurl]:
+            os.environ['realname'] = conf.config['api_host_options'][apiurl]['realname']
```
Comment (adrianschroeter) [inline osc/commandline.py]: "realname" is not a good idea for a global ENV variable, but use something like "OSC_VC_REALNAME" or alike.

Comment (Conan-Kudo) [reply osc/commandline.py]: Do we also want to change the "mailaddr" one too? It's pretty generic as well...

**Comment by Conan-Kudo** [comment]
@adrianschroeter This should be addressed now.

**Comment by Conan-Kudo** [comment]
@marcus-h Could you please review/merge this?

**Comment by jonsger** [comment]
Is there a way to disable this behaviour?

**Comment by Conan-Kudo** [comment]
@jonsger `obs-build` has a flag to filter out the name if you want the old behavior, on a per-project basis. But I don't know of a reason why you'd intentionally want the ambiguity of email-address-only.

**Comment by jonsger** [comment]
```
-------------------------------------------------------------------
Wed Nov 21 16:30:18 UTC 2018 - Jonathan Brielmaier <jbrielmaier@suse.de>
```
^ It looks a little broken, if you have a long name...

**Comment by Conan-Kudo** [comment]
@jonsger I'm curious if extending the number of dashes would actually break the format...

**Comment by lethliel** [comment]
The question is how long the line should be.

**Comment by krig** [comment]
It looks like this PR broke the `tar_scm` service which uses `$VC_MAILADDR` to get the author name:

https://github.com/openSUSE/obs-service-tar_scm/issues/296

**Comment by Conan-Kudo** [comment]
This PR sets `VC_MAILADDR`, so I don't know what the issue is?

**Comment by krig** [comment]
> This PR sets `VC_MAILADDR`, so I don't know what the issue is?

Yeah, my mistake. There is something else causing the problem.

**Pull request merged**